console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/client/device/tag/DeviceTagAddDialog.java

```diff
@@ -55,12 +55,17 @@ public void submit() {
 
         String tagId = tagsCombo.getValue().getId();
 
-        GWT_DEVICE_SERVICE.addDeviceTag(xsrfToken, selectedDevice.getScopeId(), selectedDevice.getId(), tagId, new AsyncCallback<Void>() {
+        GWT_DEVICE_SERVICE.addDeviceTag(xsrfToken, selectedDevice.getScopeId(), selectedDevice.getId(), tagId, new AsyncCallback<Boolean>() {
 
             @Override
-            public void onSuccess(Void gwtAccessPermission) {
+            public void onSuccess(Boolean gwtAccessPermission) {
                 exitStatus = true;
-                exitMessage = MSGS.dialogDeviceTagAddConfirmation();
+                if (gwtAccessPermission) {
```

Comment:
After throwing the exception on the server side, manage the error message in `onFailure` method, showing the message related to the duplicate tag when it happens or a generic message if other exception occurs